dashboard/test/models/user_test.rb

```diff
@@ -1563,6 +1563,18 @@ def track_progress(user_id, script_level, result, pairings: nil)
     assert_equal [], teacher.reload.permissions
   end
 
+  test 'assign_course_as_facilitator assigns course to facilitator' do
+    facilitator = create :facilitator
+    facilitator.course_as_facilitator = Pd::Workshop::COURSES.last
```

Comment:
Indeed, the last 3 lines are more verbose than your suggestion. I'm ok with asserting the `.any?` query (or even better, `.exists?`). I still want the `assert_creates` so we ensure that line actually creates the record.

Good point about it not creating a record if one already exists, i.e. idempotence. We should verify that in another test case :)

Set up a facilitator with a CourseFacilitator entry, then `assert_no_difference` around the assignment.